2025-02-24: finished the changes requested. see commit with message date: 2025-02-24

UselessDross opened the pull request:
## What does this PR do?
- Implements Git cloning for private repositories using personal access tokens
- Improves error handling when authentication fails
- Adds integration tests for cloning private repositories

## How was this tested?
- Ran `yarn test:int` to verify correct behavior
- Manually tested cloning public and private repos

## Related Issues
Closes #1040, Closes #1041

UselessDross committed:
Integrates GitLab Runner and updates documentation (#1115 and #1082)

  - Adds integrated gitlab runner to all installation scenarios.
    It is now possible to have one runner that can run all
    the jobs sent by DevOps requests.
  - Updates DevOps documentation and ports corrections from
    documentation of release v0.6.0

---------
Co-authored-by: aryanpingle <realaryanpingle@gmail.com>

UselessDross committed:
Merge branch 'feature/Private_git_repos' of https://github.com/UselessDross/DTaaS_FeatureBranch_ClonePrivateUsingLims into feature/Private_git_repos

UselessDross committed:
is this working?

UselessDross committed:
is this working?

UselessDross committed:
feat: add integration tests for private repo cloning in GitFilesService

UselessDross committed:
Merge branch 'feature/Private_git_repos' of https://github.com/UselessDross/DTaaS_FeatureBranch_ClonePrivateUsingLims into feature/Private_git_repos

UselessDross committed:
feat: add integration tests for private repo cloning in GitFilesService

UselessDross committed:
2025-02-16: Implemented and tested the git cloning service

UselessDross committed:
removed the files folder

prasadtalasila commented on .gitlab-ci.yml:
please remove this file

prasadtalasila commented on client/.prettierrc:
please undo the changes to this file

prasadtalasila commented on package-lock.json:
please remove this file

prasadtalasila commented on servers/.gitignore:
please move this file to `servers/lib` folder

prasadtalasila commented on servers/lib/config/libms.dev.yaml:
please change the contents of this file to the following:

```yaml
port: '4001'
mode: 'local'    #git or local
local-path: '/dtaas/libms/files'
log-level: 'debug'
apollo-path: '/lib'
graphql-playground: 'true'

git-repos:   #only used in git mode
  - user1: 
      repo-url: 'https://gitlab.com/dtaas/user1.git'
      http-token: 'xxxx'
  - user2: 
      repo-url: 'https://gitlab.com/dtaas/user2.git'
      http-token: 'xxxx'
  - common:
      repo-url: 'https://gitlab.com/dtaas/common.git'
      http-token: 'xxxx'
```

prasadtalasila commented on servers/lib/config/libms.dev.yaml:
Please copy the new contents of `libms.dev.yaml` to `libms.yaml.default`

prasadtalasila commented on servers/lib/libms.yaml:
please remove this file

prasadtalasila commented on servers/lib/package-lock.json:
this should be removed

prasadtalasila commented on servers/lib/package.json:
please update the version number on line-3 to `0.5.5`

prasadtalasila commented on servers/lib/package.json:
```diff
@@ -14,6 +14,7 @@
   "type": "module",
   "main": "dist/src/main.js",
   "scripts": {
+    "lint": "eslint . ",
```
remove this line

prasadtalasila commented on servers/lib/src/config/config.service.ts:
```diff
@@ -1,9 +1,12 @@
+import { ConfigValues } from './config.model.js';
 import resolveFile from './util.js';
 import { IConfig } from './config.interface.js';
+import dotenv from 'dotenv';
```
please remove `.env` code

prasadtalasila commented on servers/lib/src/files/git/git-files.service.ts:
```diff
@@ -20,46 +20,65 @@ export default class GitFilesService implements IFilesService {
 
-  private cloneRepositories(): Promise<void[]> {
-    const userRepoConfigs = this.configService.getGitRepos();
+  private buildAuthedUrl(repoUrl_: string, httpToken_?: string): string { return httpToken_ ? `https://${httpToken_}@${repoUrl_.replace('https://', '')}` : repoUrl_; }
```
please rename the function to `buildAuthUrl`

prasadtalasila reviewed (commented):
@UselessDross thanks for the PR. Please check the suggested changes.

UselessDross committed:
2025-02-24:
> fixed unnessesary comments,
- Removed uneessesary comments.
- removed the unessesary indent in the '.prittierrc' file(s)

+ added a util folder at: 'servers/lib/src' that contains all new utility functions.
+ added a Color logger class.
+ used the new class in the 'git-files.service.ts'
+ in 'servers/lib/src/bootstrap.ts' changed line 17 to use and search for the 'libms.dev.yaml' at its reletive location.

? changed the config files to use the 'http-token' type.

UselessDross committed:
Merge branch 'INTO-CPS-Association:feature/distributed-demo' into feature/distributed-demo

UselessDross committed:
chekking if this owrks

UselessDross commented:
## on git commit 3187a34400f2ba20f446fbc33249b0e53ff4c671

finished the following tasks:

- [?] in the config file, we are loading the http ural for the git repo,
         and also load the http-token along side it, so we can use them 
         in cloning later.
         in the ->config.service.ts<- look at the old on before your commits. 
         
         in that git getGitrepos must load two things, one are the tokens 
         and the other is the url.  

         if you look at config.model.ts, at 
         line 12 it is loifng the url as a type string, but you should add 
         the loading of the type http-token.

         so you need to modify all the fils in config directory. the 
         reason is, the git repo type is used in all the files, so you 
         need to make the change in the whole folder.


- [V] remove the '//';
- [V] remove the // NOTE comments; because comments are seserved for the most complex part of the code.
- [?] chalk. a better coloring options to the terminal. that way the code becomes easier to read and understand what does what.
- [V] create in servers/lib/src a folder calld util. and then put your loggoer funtion there, for your error or consol information. there make the logger funtion.
- [V] as soon as you are finished. email mr. Talasila;

prasadtalasila commented on servers/lib/config/libms.dev.yaml:
please revert changes to the first six lines of this file

prasadtalasila commented on servers/lib/config/libms.yaml.default:
again, please make this file same as `libms.yaml`

UselessDross replied:
I removed the libms.yaml file, but if you want this file to be the same as 
the libms.yaml file? then here it is, i think?
Do please tell me when im wrong, please.

prasadtalasila commented on servers/lib/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@into-cps-association/libms",
-  "version": "0.5.4",
+  "version": "0.5.5",
```
this would be **0.6.0**

UselessDross replied:
Im not sure if it is supposed to be 0.6.0 or 0.5.5?

prasadtalasila commented on servers/lib/src/bootstrap.ts:
```diff
@@ -14,7 +14,7 @@ export default async function bootstrap(options?: BootstrapOptions) {
 
   if (process.env.LIBMS_CONFIG_PATH == null)
-    process.env.LIBMS_CONFIG_PATH = options.config ?? 'libms.yaml';
+    process.env.LIBMS_CONFIG_PATH = options.config ?? '../lib/config/libms.dev.yaml';
```
why is this changed?

UselessDross replied:
The change here is because i removed the 'libms.yaml' file so
logically, i think, it should default to where the other '.yaml' file.
And that file is in the location, shown by the relative path.

prasadtalasila commented on servers/lib/src/util/Interfaces/IConsoleLogger.ts:
The [NestJS Logger](https://docs.nestjs.com/techniques/logger) has interface and functions with certain naming conventions. Please comply with these conventions. In fact, you can create an overloaded logger. Please see if you can use this technique.

UselessDross replied:
I did not know that. I thought i was following the naming convention?
And i thought i was following the SOLID principles?
i will look into it.

prasadtalasila commented on servers/lib/src/files/git/git-files.service.ts:
```diff
@@ -1,65 +1,79 @@
+import * as git from 'isomorphic-git';
+import * as fs from 'fs';
 import path from 'path';
+const _logger = new ConsoleLogger();
```
This is not needed if you use overloaded NestJS logger. Even if you need to create a new logger, please keep it inside the **GitFilesService** class.

UselessDross replied:
Sorry, i guess i should have given it another name😅.
For context, i don't actually try or intent to overload the 'NestJS logger'.
im not sure what to name it🤔?

prasadtalasila commented on servers/lib/yarn.lock:
please revert changes to this file

UselessDross replied:
Like this?
I was not aware this file got changed.

prasadtalasila commented on servers/lib/test/integration/git-files.integration.spec.ts:
```diff
@@ -0,0 +1,71 @@
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+dotenv.config();
```
we are not using .env

prasadtalasila commented on servers/lib/test/integration/git-files.integration.spec.ts:
```diff
@@ -0,0 +1,71 @@
+    private config: any;
+    constructor(configData: any) { this.config = configData; }
+    getLocalPath(): string { return this.config['local-path']; }
+    getGitRepos(): any[] { return this.config['git-repos']; }
```
please avoid any

UselessDross replied:
not sure, what type to give the
'private config'
i would change it to use the 'ConfigValues' type, but then i would have
to change the 'getGitRepos()'.
so im not sure here?

prasadtalasila commented on servers/lib/test/integration/git-files.integration.spec.ts:
This testing code does not follow the recommended NestJS testing patterns. Please see [file service](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/servers/lib/test/integration/files.service.integration.spec.ts) testing example. You can also check [this example](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/servers/execution/runner/test/integration/execa-manager.service.spec.ts).

prasadtalasila reviewed (commented):
@UselessDross Thanks for the updates. Please see the PR comments. In addition,

1. Please resolve the code issues identified by Codacy
2. Revert all the changes in `files` directory

UselessDross committed:
2025-02-25: supposedly finsihed the comments recived

UselessDross committed:
2025-02-26:
%  Changed line 39 in file 'git-file.integration.spec.ts' at dirctory 'servers/lib/test/integration/'

UselessDross committed:
2025-02-26v2:
- removed the unnesseary condition at line 39

% Changed the line 'const userRepoConfigs = this.configService.getGitRepos()';
% Changed the catch block on line 69;
% Changed in 'servers/lib/integration/git-files.integration.spec.ts' all occurcences of ConfigValue datatype to just eh GitRepo datatype sinces that seems to be the nessesary one;

+ Added the use of the 'ConsolLogger' object from the utility folder.

UselessDross committed:
is this owrking?

UselessDross committed:
New commit to check if this solved the testing problem

UselessDross committed:
When looking at codacy, for some reason codacy production prevents the last bit of the codebase to pass. the reson for this seems to be due to strict formatting and styling that codacy expects. therefore, having little to no clue what there is to be done, i tried adding the expected formatting/styling rules to the .pretterrc file at servers/lib, in hope this would solve the issue.

UselessDross committed:
okay i removed it.

UselessDross committed:
Just trying something out. will revert if it does not work.

UselessDross committed:
it didn't work

UselessDross committed:
2025-03-03:
Implement ConsoleLogger to follow NestJS style and integrate into GitFilesService tests

UselessDross committed:
Refactor ConsoleLogger: Simplify constant declarations and remove unused constructor

prasadtalasila commented:
@UselessDross the consolidated list of suggestions for this PR:

1) Please do not changes files outside of `servers/lib` directory. The PR #1562 shows an example of files that can change in this PR. The changes dempnstrated in PR #1562 are based on the code changes you made before last Thursday. So, they are somewhat outdated but the suggestion of limiting changes to `servers/lib` is still valid.
2) Please address the suggestions previously given by me in this PR.
3) Please address the code issues identified by Codacy. You can click on the Codacy checks to see the list of issues. 
4) Please follow nestjs conventions in the new code contributions.

UselessDross committed:
2025-03-04: trying to fix the codacy issues.
Refactor GitFilesService: Improve error handling and logging, ensure valid configurations

UselessDross committed:
Revert all folders and files except servers to commit abc123

UselessDross committed:
revert servers folder to commit 3b553f9dad53c11fbedcb32e238b557fdc14e077

UselessDross committed:
should now have reverted the changeds to the files/ folder

UselessDross committed:
test

UselessDross committed:
Add UserName property to GitRepo type and update GitFilesService for optional repo configuration

UselessDross committed:
Refactor GitFilesService to return Promise<void> in cloneRepositories and init methods

UselessDross committed:
Remove UserName property from GitRepo type and refactor cloneRepositories method for improved promise handling

UselessDross committed:
Add validation for user keys in cloneRepositories method

UselessDross committed:
Refactor GitRepo type to interface and enhance cloneRepositories method with additional validation for repo configuration

prasadtalasila commented:
Further development will happen on Khaled's fork.
